File: linear_algebra/C++/NDTh_BT8.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cout << " nhap n:";
    cin >> n;
    if (n <= 0 ) {
        cout << " nhap lai n fen oiii:";
        return 0;
    }
    cout << "Ma tran 0:\n";
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) cout << 0 << " ";
        cout << endl;
    }

    cout << "\nMa tran don vi:\n";
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++)
            cout << (i == j ? 1 : 0) << " ";
        cout << endl;
    }
}

File: linear_algebra/Python/NDTh_BT8.py
n = int(input("nhap n:"))

if n <= 0:
    print("nhap lai n di fen:))")
else:
    print("Ma tran 0:")
    for i in range(n):
        print("0 " * n)

    print("Ma tran don vi:")
    for i in range(n):
        for j in range(n):
            print(1 if i == j else 0, end=" ")
        print()
